src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -28,15 +27,16 @@ internal sealed class MsQuicStream : QuicStreamProvider
         private bool _started;
 
         // Used by the class to indicate that the stream is m_Readable.
-        private readonly bool _canRead;
+        private bool _canRead;
```

Comment:
Do we really need this? Wouldn't be enough to use the `ReadState`? Potentially expand it for what `Dispose` needs. Same for `_canWrite`.

Reply:
We probably could. If we want to replaces existing bools with state compassion I would do it as separate change. For now, we have conformance tests expecting this to become `false` on dispose/shutdown so I did minimal work to get expected behavior.

Reply:
Fair enough. In case we clean this up, we could just combine the readonly `_canRead` with `ReadState` in `CanRead`, e.g.: `CanRead => _canRead && ReadState != Closed;`

Reply:
I agree with @ManickaP here. It seems less than ideal to change this now with the intent to change back in the future. @wfurt how hard is it to implement the suggestion from @ManickaP?

Reply:
updated. I have less confidence in the state machine than the bool but so be it. This is not critical for the PR and I did the change to pass more conformance tests. That can possibly be done separately.